src/client.rs

```diff
@@ -1453,6 +1453,19 @@ impl<C: Config> Client<C> {
                 let entity = entities
                     .get(id)
                     .expect("entity IDs in spatial index should be valid at this point");
+
+                // We want to skip player entities that are not in the players player_list.
+                if entity.kind() == EntityKind::Player {
+                    if let Some(list_id) = &self.new_player_list {
+                        let entry = player_lists.get(list_id).get_entry(&entity.uuid());
+
+                        #[allow(clippy::question_mark)]
```

Comment:
would it be possible to not ignore this lint?

Reply:
Unfortunately not. Adding a question mark as suggested would change the implementation. We don't want to return the entity if it is found. Only return early if it is not. There is a bugfix in clippy that should come with the next release.